x/stake/types/pool.go

```diff
@@ -52,108 +42,36 @@ func InitialPool() Pool {
 }
 
-// get the exchange rate of bonded token per issued share
-func (p Pool) BondedShareExRate() sdk.Rat {
-	if p.BondedShares.IsZero() {
-		return sdk.OneRat()
-	}
-	return sdk.NewRat(p.BondedTokens).Quo(p.BondedShares)
-}
-
-// get the exchange rate of unbonding tokens held in validators per issued share
-func (p Pool) UnbondingShareExRate() sdk.Rat {
-	if p.UnbondingShares.IsZero() {
-		return sdk.OneRat()
-	}
-	return sdk.NewRat(p.UnbondingTokens).Quo(p.UnbondingShares)
-}
-
-// get the exchange rate of unbonded tokens held in validators per issued share
-func (p Pool) UnbondedShareExRate() sdk.Rat {
-	if p.UnbondedShares.IsZero() {
-		return sdk.OneRat()
-	}
-	return sdk.NewRat(p.UnbondedTokens).Quo(p.UnbondedShares)
-}
-
 //_______________________________________________________________________
 
-func (p Pool) addTokensUnbonded(amount int64) (p2 Pool, issuedShares PoolShares) {
-	issuedSharesAmount := sdk.NewRat(amount).Quo(p.UnbondedShareExRate()) // tokens * (shares/tokens)
-	p.UnbondedShares = p.UnbondedShares.Add(issuedSharesAmount)
-	p.UnbondedTokens += amount
-	p.LooseTokens -= amount
-	if p.LooseTokens < 0 {
-		panic(fmt.Sprintf("sanity check: loose tokens negative, pool: %v", p))
-	}
-	return p, NewUnbondedShares(issuedSharesAmount)
-}
-
-func (p Pool) removeSharesUnbonded(shares sdk.Rat) (p2 Pool, removedTokens int64) {
-	removedTokens = p.UnbondedShareExRate().Mul(shares).RoundInt64() // (tokens/shares) * shares
-	p.UnbondedShares = p.UnbondedShares.Sub(shares)
-	p.UnbondedTokens -= removedTokens
-	p.LooseTokens += removedTokens
-	if p.UnbondedTokens < 0 {
-		panic(fmt.Sprintf("sanity check: unbonded tokens negative, pool: %v", p))
-	}
-	return p, removedTokens
-}
-
-func (p Pool) addTokensUnbonding(amount int64) (p2 Pool, issuedShares PoolShares) {
-	issuedSharesAmount := sdk.NewRat(amount).Quo(p.UnbondingShareExRate()) // tokens * (shares/tokens)
-	p.UnbondingShares = p.UnbondingShares.Add(issuedSharesAmount)
-	p.UnbondingTokens += amount
-	p.LooseTokens -= amount
-	if p.LooseTokens < 0 {
-		panic(fmt.Sprintf("sanity check: loose tokens negative, pool: %v", p))
-	}
-	return p, NewUnbondingShares(issuedSharesAmount)
-}
-
-func (p Pool) removeSharesUnbonding(shares sdk.Rat) (p2 Pool, removedTokens int64) {
-	removedTokens = p.UnbondingShareExRate().Mul(shares).RoundInt64() // (tokens/shares) * shares
-	p.UnbondingShares = p.UnbondingShares.Sub(shares)
-	p.UnbondingTokens -= removedTokens
-	p.LooseTokens += removedTokens
-	if p.UnbondedTokens < 0 {
-		panic(fmt.Sprintf("sanity check: unbonding tokens negative, pool: %v", p))
-	}
-	return p, removedTokens
-}
-
-func (p Pool) addTokensBonded(amount int64) (p2 Pool, issuedShares PoolShares) {
-	issuedSharesAmount := sdk.NewRat(amount).Quo(p.BondedShareExRate()) // tokens * (shares/tokens)
-	p.BondedShares = p.BondedShares.Add(issuedSharesAmount)
-	p.BondedTokens += amount
-	p.LooseTokens -= amount
-	if p.LooseTokens < 0 {
+func (p Pool) addBondedTokens(bondedTokens sdk.Rat) Pool {
```

Comment:
Rename `looseTokensToBonded` perhaps?

Reply:
yeah that's more concise - good idea - changed